pkg/lint/lintersdb/builder_linter.go

```diff
@@ -0,0 +1,712 @@
+package lintersdb
+
+import (
+	"github.com/golangci/golangci-lint/pkg/config"
+	"github.com/golangci/golangci-lint/pkg/golinters"
+	"github.com/golangci/golangci-lint/pkg/lint/linter"
+)
+
+// LinterBuilder builds the "internal" linters based on the configuration.
+type LinterBuilder struct{}
+
+// NewLinterBuilder creates a new LinterBuilder.
+func NewLinterBuilder() *LinterBuilder {
+	return &LinterBuilder{}
+}
+
+// Build loads all the "internal" linters.
+// The configuration is use for the linter settings.
+func (b LinterBuilder) Build(cfg *config.Config) []*linter.Config {
+	if cfg == nil {
+		return nil
+	}
+
+	const megacheckName = "megacheck"
+
+	// The linters are sorted in the alphabetical order (case-insensitive).
+	// When a new linter is added the version in `WithSince(...)` must be the next minor version of golangci-lint.
+	return []*linter.Config{
```

Comment:
Looks like we should update https://github.com/golangci/golangci-lint/blob/9c492afc8396d16bb640661afecf8e2968f04c60/docs/src/docs/contributing/architecture.mdx#L25C1-L48C1

Reply:
Clearly no, the package loading hasn't changed, it's offtopic.